homeassistant/components/permobil/config_flow.py

```diff
@@ -0,0 +1,217 @@
+    data: dict[str, str] = {
+        CONF_EMAIL: "",
+        CONF_REGION: "",
+        CONF_CODE: "",
+        CONF_TOKEN: "",
+        CONF_TTL: "",
+    }
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Invoke when a user initiates a flow via the user interface."""
+        errors: dict[str, str] = {}
+        if not self.p_api:
+            # create the api instance to use for validation of the user input
+            session = async_get_clientsession(self.hass)
+            self.p_api = MyPermobil(APPLICATION, session=session)
+
+        try:
+            if user_input is not None:
+                if not user_input.get(CONF_EMAIL):
+                    raise InvalidAuth("empty email")
+
+                # the user has entered their email in the first prompt
+                user_input[CONF_EMAIL] = user_input[CONF_EMAIL].replace(" ", "")
+                await validate_input(self.p_api, user_input)  # ClientException
+                self.data[CONF_EMAIL] = user_input[CONF_EMAIL]
+                _LOGGER.debug("Permobil: email %s", self.p_api.email)
+
+                await self.async_set_unique_id(self.data[CONF_EMAIL])
+                self._abort_if_unique_id_configured()
+        except MyPermobilClientException as err:
+            # the email did not pass validation by the api client
+            _LOGGER.error("Permobil: %s", err)
+            errors["base"] = f"Pemobil: {err}"
+            errors["reason"] = "invalid_email"
+        except InvalidAuth as err:
+            _LOGGER.error("Permobil: %s", err)
+            errors["base"] = "Empty Email"
+            errors["reason"] = "empty_email"
+
+        if errors or user_input is None:
+            # There was an error when the user entered their email
+            # or the user opened the flow for the first time
+            return self.async_show_form(
+                step_id="user", data_schema=GET_EMAIL_SCHEMA, errors=errors
+            )
+        # email prompt finished successfully, open the select region prompt
+        return await self.async_step_region()
+
+    async def async_step_region(self, user_input=None) -> FlowResult:
+        """Invoke when a user initiates a flow via the user interface."""
+        errors: dict[str, str] = {}
+        if not self.p_api:
+            # create the api for getting regions and sending the code
+            session = async_get_clientsession(self.hass)
+            self.p_api = MyPermobil(APPLICATION, session=session)
+
+        try:
+            if user_input is None:
+                # The user has opened the 2nd prompt for the first time
+                # if the email ends with @permobil,
+                # include internal regions for debugging purposes
+                include_internal = self.data[CONF_EMAIL].endswith("@permobil.com")
+                _LOGGER.debug("Permobil: include internals %s", include_internal)
```

Comment:
```suggestion
```

We don't accept this code. Permobil should check at their server if the email ends with `"@permobil.com"` and include then debug instances